### PR TITLE
Interval analysis fix

### LINKED ISSUE #538: Interval Analysis is always generating inductive instructions
This is causing conflicts when running esbmc with incremental approach.

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
This should close #538

### CODE REVIEW

File: src/goto-symex/execution_state.cpp [hunk 1]
```diff
@@ -239,6 +239,10 @@ void execution_statet::symex_step(reachability_treet &art)
   // case or forward condition
   if((base_case || forward_condition) && instruction.inductive_step_instruction)
   {
+    // This assertion will prevent us of having weird side-effects (issue #538)
+    assert(
+      k_induction &&
+      "Inductive step instructions should be setted only for k-induction");
```
Comment (lucasccordeiro) [inline src/goto-symex/execution_state.cpp]: "...should be setted only for k-induction" -> "...should be set only for k-induction"

File: src/goto-symex/execution_state.cpp [hunk 2]
```diff
@@ -239,6 +239,10 @@ void execution_statet::symex_step(reachability_treet &art)
   // case or forward condition
   if((base_case || forward_condition) && instruction.inductive_step_instruction)
   {
+    // This assertion will prevent us of having weird side-effects (issue #538)
```
Comment (lucasccordeiro) [inline src/goto-symex/execution_state.cpp:242]: What do you mean by "weird side-effects"?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]

File: src/goto-symex/execution_state.cpp (hunk 2 above)
Comment (rafaelsamenezes) [reply src/goto-symex/execution_state.cpp:242]: Skipping instructions during incremental. https://github.com/esbmc/esbmc/issues/538

Comment (lucasccordeiro) [reply src/goto-symex/execution_state.cpp:242]: Can you say that in the comment?

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

**Review by mikhailramalho — CHANGES_REQUESTED** [review]
Nice catch, but you don't really need to pass that boolean to `interval_analysis()`, simply call `config.options.is_kind()` when needed

**Pull request merged**